src/Microsoft.Data/DataFrameColumnContainerBinaryOperations.cs

```diff
@@ -0,0 +1,162 @@
+﻿
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.Data
+{
+    public partial class PrimitiveDataFrameColumnContainer<T>
```

Comment:
Can you keep the file names and the type that is contained in the file the same? That way it is easier to find classes when navigating the file system.

For partial classes, a typical pattern is use put a `.` in the file name to separate type name from the category of things in it. So this file could be called `PrimitiveDataFrameColumnContainer.BinaryOperations.cs`.